### PR TITLE
Add option to disable the `ServiceAccount` creation for `teleport-kube-agent chart`

### PR DESCRIPTION
**Pull request by tigrato**
When assigning AWS IAM permissions to Teleport, `eksctl create iamserviceaccount` command manages the ServiceAccount lifecycle. 

This PR allows any user to configure an existing Kubernetes ServiceAccount used by the agent.

Part of #11866

### CODE REVIEW

**Comment by hugoShaka** [comment]
Hey, I'm sorry: I forgot to update the `teleport-kube-agent` chart when merging https://github.com/gravitational/teleport/pull/16337

Could you use the same value names than the `teleport-cluster` chart so both charts are aligned?

After checking quickly what the Helm community is doing it seems most modern charts converged to the signature

```
serviceAccount:
  create: true
  name: ""
  # ...
```

**Review by hugoShaka — CHANGES_REQUESTED** [review]
Hey, I'm sorry: I forgot to update the `teleport-kube-agent` chart when merging https://github.com/gravitational/teleport/pull/16337

Could you use the same value names than the `teleport-cluster` chart so both charts are aligned?

After checking quickly what the Helm community is doing it seems most modern charts converged to the signature

```
serviceAccount:
  create: true
  name: ""
  # ...
```

I know the name is already managed by `serviceAccountName`, a workaround would be to coalesce the values.

File: examples/chart/teleport-kube-agent/tests/serviceaccount_test.yaml [hunk 1]
```diff
@@ -25,3 +25,9 @@ tests:
           path: metadata.annotations.kubernetes\.io/serviceaccount-different
           value: 5
       - matchSnapshot: {}
+  - it: sets ServiceAccount labels when specified
```
Comment (hugoShaka) [inline examples/chart/teleport-kube-agent/tests/serviceaccount_test.yaml:28]: Copy-paste error

**Review by hugoShaka — APPROVED** [review]

**Review by probakowski — APPROVED** [review]

**Review by alexfornuto — APPROVED** [review]

**Comment by alexfornuto** [comment]
@tigrato the backporting for the PR is failing the [linter](https://console.cloud.google.com/cloud-build/builds;region=us-west1/e25bb59e-3bef-48b5-888f-7de068d02721?project=ci-account) checks.

**Pull request merged**